### PR TITLE
Add documentation of Kani result types

### LINKED ISSUE #863: Missing explanation of result types in docs
The documentation currently lacks an explanation of the result types that Kani produces (SUCCESS, FAILURE, UNREACHABLE, etc.)

### PR DESCRIPTION
**Pull request by zhassan-aws**
### Description of changes: 

Document the different results a check can have in Kani (SUCCESS, FAILURE, UNDETERMINED, UNREACHABLE).

### Resolved issues:

Resolves #863 

### Testing:

* How is this change tested? New doc tests.

* Is this a refactor change? No

### Checklist
- [X] Each commit message has a non-empty body, explaining why the change was made
- [X] Methods or procedures are documented
- [X] Regression or unit tests are included, or existing tests cover the modified code
- [X] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: docs/src/kani-result-types.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+```
+Check 4: success_example.assertion.4
+         - Status: SUCCESS
+         - Description: "assertion failed: sum == 6"
+```
+
+2. `FAILURE`: This indicates that the check failed, i.e., the property doesn't
+hold. In this case, you can view the failure trace by re-running with the
+`--visualize` option.  This generates a `report` directory that contains the
+trace information. Load the `report/html/index.html` file in a browser to
```
Comment (tedinski) [inline docs/src/kani-result-types.md:24]: The new cargo-kani prints where the report was generated, and this is under `target/` for `cargo-kani` vs `kani`, so I recommend not stating the path here.

Comment (zhassan-aws) [reply docs/src/kani-result-types.md]: Done.

File: docs/src/kani-result-types.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    5. Divide/remainder-by-zero checks
+
+
+Example:
+
+```rust
+{{#include getting-started/result-types/src/main.rs:unreachable_example}}
+```
+
+The output from Kani when run with `--assertion-reach-checks` indicates that
```
Comment (tedinski) [inline docs/src/kani-result-types.md:56]: IMO we should emphasize this flag more, right now it seems a bit buried that this flag has to be enabled to get this result. I think burying the flag down here only makes sense if it's no by default, which it isn't yet right?

 Maybe by adding "(requires `--assertion-reach-checks`)" after UNREACHABLE above?

**Review by tedinski — APPROVED** [review]
One little thing, otherwise lgtm.

File: docs/src/kani-result-types.md (hunk 2 above)
Comment (zhassan-aws) [reply docs/src/kani-result-types.md:56]: Makes sense. Added.

**Pull request merged**